src/main/java/com/accantosystems/stratoss/vnfmdriver/web/etsi/GrantController.java

```diff
@@ -71,13 +71,22 @@ public ResponseEntity<Grant> requestGrant(@RequestBody GrantRequest grantRequest
     public ResponseEntity<Grant> getGrant(@PathVariable String grantId) throws GrantRejectedException, GrantProviderException {
         logger.info("Received grant fetch for id [{}]", grantId);
         UUID uuid = UUID.randomUUID();
-        LoggingUtils.logEnabledMDC(grantId, MessageType.REQUEST, MessageDirection.RECEIVED, uuid.toString(),MediaType.APPLICATION_JSON.toString(), "https",getRequestProtocolMetaData(GRANT_LOCATION) ,null);
-        Grant grant = grantService.getGrant(grantId);
-        LoggingUtils.logEnabledMDC(grant != null ? grant.toString() : null, MessageType.RESPONSE,MessageDirection.SENT,grantId ,MediaType.APPLICATION_JSON.toString(), "https",null,null);
-        if (grant != null) {
-            return ResponseEntity.ok(grant);
-        } else {
-            return ResponseEntity.accepted().build();
+        try {
+            Grant grant = grantService.getGrant(grantId);
+            LoggingUtils.logEnabledMDC(grantId, MessageType.REQUEST, MessageDirection.RECEIVED, uuid.toString(), MediaType.APPLICATION_JSON.toString(), "https", getRequestProtocolMetaData(GRANT_LOCATION), null);
+            LoggingUtils.logEnabledMDC(grant != null ? grant.toString() : null, MessageType.RESPONSE, MessageDirection.SENT, grantId, MediaType.APPLICATION_JSON.toString(), "https", null, null);
+            if (grant != null) {
+                LoggingUtils.logEnabledMDC(grantId, MessageType.REQUEST, MessageDirection.RECEIVED, uuid.toString(), MediaType.APPLICATION_JSON.toString(), "https", getRequestProtocolMetaData(GRANT_LOCATION), grant.getVnfLcmOpOccId());
```

Comment:
Line 76 & 77 covers both if & else blocks of the code. Why do we need logging again in if & else blocks?